File: Day4/model/SectionRangePair.cs
namespace Day4.Model;

public class SectionRangePair
{
    private SectionRange _first { get; }
    private SectionRange _second { get; }

    private SectionRangePair(SectionRange first, SectionRange second)
    {
        _first = first;
        _second = second;
    }

    public static SectionRangePair Create(SectionRange first, SectionRange second) => 
        new SectionRangePair(first, second);

    public bool SectionsOverlap(Overlap overlapType) =>
        _first.OverlapsWith(_second, overlapType) || _second.OverlapsWith(_first, overlapType);
}

File: Day4_Java/src/model/SectionRangePair.java
package src.model;

import src.model.types.Overlap;

public class SectionRangePair {
    private SectionRange first;
    private SectionRange second;

    public SectionRangePair(SectionRange first, SectionRange second) {
        this.first = first;
        this.second = second;
    }

    public boolean SectionsOverlap(Overlap overlapType) {
        return first.OverlapsWith(second, overlapType) || second.OverlapsWith(first, overlapType);
    }
}
